tests/stdlib/thttpclient_standalone.nim

```diff
@@ -21,3 +21,34 @@ block: # bug #16436
   asyncCheck startServer()
   doAssertRaises(ProtocolError):
     waitFor runClient()
+
+block: # bug #14794 (And test for presence of content-length header when using postContent)
+  proc startServer() {.async.} =
+    var killServer = false
+    proc cb(req: Request) {.async.} =
+      doAssert(req.body.endsWith(httpNewLine), "Multipart body does not end with a newline.")
+      # this next line is probably not required because asynchttpserver does not call
+      # the callback when there is no content-length header.  It instead errors with 
+      # Error: unhandled exception: 411 Length Required
+      # Added for good measure in case the server becomes more permissive.
+      doAssert(req.headers.hasKey("content-length"), "Content-Length header is not present.")
+      killServer = true
+      asyncCheck req.respond(Http200, "OK")
+
+    var server = newAsyncHttpServer()
+    server.listen(Port(5556))
```

Comment:
@mp035 can you send a followup PR to improve this test by avoiding hardcoded ports? these cause flaky failures, see https://github.com/timotheecour/Nim/issues/534#issuecomment-821904874

Reply:
Will do.